src/dftbp/transport/negfint.F90

```diff
@@ -135,6 +142,20 @@ subroutine TNegfInt_init(this, transpar, env, greendens, tundos, tempElec, coord
       end do
     end do
 
+    if (isPeriodic .and. nCont > 2) then
+      allocate(sigma(min(nCont,3)))
+      allocate(contVectors(3, nCont))
+      allocate(U(3, 3))
+      allocate(Vt(nCont, nCont))
+      do iCont = 1, nCont
+        contVectors(:, iCont) = transpar%contacts(iCont)%lattice
+      end do
+      call gesvd(contVectors, U, sigma, Vt)
+      if (all(sigma > transpar%contactLayerTol**2)) then
```

Comment:
Could you factor that out into a function ˙hasFullySurroundingContacts` or something similar? Then we would not need all the auxiliary variables in the local scope.

Reply:
Sure, done.